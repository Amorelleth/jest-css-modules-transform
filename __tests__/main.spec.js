const cssFile = require('./source/style.css');
const stylusFile = require('./source/style.styl');
const sassFile = require('./source/style.sass');
const scssFile = require('./source/style.scss');
const lessFile = require('./source/style.less');
const cssNestedFile = require('./source/style-nested.css');

describe('jest-css-modules', () => {
    [
        ['css', cssFile],
        ['styl', stylusFile],
        ['sass', sassFile],
        ['scss', scssFile],
        ['less', lessFile],
    ].forEach(([ext, style]) => {
        it(`should import .${ext}`, () => {
            expect(style).toMatchSnapshot();
        });

        it(`should import .${ext} and keyframes should be last`, () => {
            const keys = Object.keys(style);

            expect(keys[0]).toBe('class1');
            // keyframes should be last, like Webpack did it
            expect(keys[keys.length - 1]).toBe('sunrise');
        });
    });

    it('should import .css with nested', () => {
        expect(cssNestedFile).toEqual({
            foo: 'foo',
            foo__bar: 'foo__bar',
            foo__zoo: 'foo__zoo',
            boo: 'boo',
        });
    });

    it('should have correct object params', () => {
        expect(typeof cssFile).toBe('object');
        expect(cssFile).toBeInstanceOf(Object);
        expect(Object.keys(cssFile)).toMatchSnapshot();
    });
});
