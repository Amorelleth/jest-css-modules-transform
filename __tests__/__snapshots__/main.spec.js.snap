// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-css-modules should have correct object params 1`] = `
Array [
  "class1",
  "class2",
  "class3",
  "id",
  "id2",
  "sunrise",
]
`;

exports[`jest-css-modules should import .css 1`] = `
Object {
  "class1": "class1",
  "class2": "class2",
  "class3": "class3",
  "id": "id",
  "id2": "id2",
  "sunrise": "sunrise",
}
`;

exports[`jest-css-modules should import .less 1`] = `
Object {
  "class1": "class1",
  "class2": "class2",
  "class3": "class3",
  "id": "id",
  "id2": "id2",
  "sunrise": "sunrise",
}
`;

exports[`jest-css-modules should import .sass 1`] = `
Object {
  "class1": "class1",
  "class2": "class2",
  "class3": "class3",
  "id": "id",
  "id2": "id2",
  "sunrise": "sunrise",
}
`;

exports[`jest-css-modules should import .scss 1`] = `
Object {
  "class1": "class1",
  "class2": "class2",
  "class3": "class3",
  "id": "id",
  "id2": "id2",
  "sunrise": "sunrise",
}
`;

exports[`jest-css-modules should import .styl 1`] = `
Object {
  "class1": "class1",
  "class2": "class2",
  "class3": "class3",
  "id": "id",
  "id2": "id2",
  "sunrise": "sunrise",
}
`;
